ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -552,10 +552,21 @@ public func mergeMap<T, U>(transform: T -> SignalProducer<U>)(producer: SignalPr
 public func switch<T>(producer: SignalProducer<SignalProducer<T>>) -> SignalProducer<T>
 public func switchMap<T, U>(transform: T -> SignalProducer<U>)(producer: SignalProducer<T>) -> SignalProducer<U>
 
-public func repeat<T>(count: Int)(producer: SignalProducer<T>) -> SignalProducer<T>
 public func retry<T>(count: Int)(producer: SignalProducer<T>) -> SignalProducer<T>
 */
 
+/// Repeats `producer` `count` times.
+/// Repeating `0` times results in a equivalent signal producer.
+public func repeat<T, E>(count: Int)(producer: SignalProducer<T, E>) -> SignalProducer<T, E> {
+	precondition(count >= 0)
+
+	if count == 0 {
+		return producer
+	} else {
+		return producer |> concat(producer |> repeat(count - 1))
```

Comment:
This is clever, but for a huge count (e.g., `Int.max`) will assumably blow the stack, since `repeat()` recursively invokes itself immediately.

Can you refactor this to use iteration instead, and add a test that verifies that an overflow doesn't occur with an `Int.max` repeat?